sdk/azidentity/device_code_credential.go

```diff
@@ -15,27 +16,51 @@ const (
 	deviceCodeGrantType = "urn:ietf:params:oauth:grant-type:device_code"
 )
 
+// DeviceCodeCredentialOptions provide options that can configure DeviceCodeCredential instead of using the default values.
+type DeviceCodeCredentialOptions struct {
+	// Gets the Azure Active Directory tenant (directory) ID of the service principal
+	TenantID string
+	// Gets the client (application) ID of the service principal
+	ClientID string
+	// The callback function used to send the login message back to the user
+	UserPrompt func(string)
```

Comment:
Is UserPrompt a clear enough name for the callback?

Reply:
I was thinking about this.  Maybe just `Prompt`?  What do other SDKs call this?

Reply:
Well there's a mix of names for the other langs:
NET - deviceCodeCallback
JAVA - challengeConsumer
JS/TS - userPromptCallback
PYTHON - prompt_callback

Reply:
Leave as `UserPrompt` for now per discussion.